src/components/tooltip/tooltip.jsx

```diff
@@ -19,58 +14,71 @@ export default function Tooltip({
   disabled,
   id,
   text,
-  truncate,
 }) {
-  const [visible, setVisible] = useState(false);
+  // size of the triangle in pixels
+  const triangleHeight = 8;
 
-  if (disabled) return children;
+  const tooltipRef = useRef();
+  const { position, visible, show, hide } = useTooltipPositioning({ tooltipRef, triangleHeight, direction });
 
-  const show = () => setVisible(true);
-  const hide = () => setVisible(false);
+  useLayoutEffect(() => {
+    if (disabled || document.querySelector('[aria-describedby]', id)) return;
 
-  const classNames = [className];
+    throw new Error('<Tooltip> was used without an element on the DOM being described by it. ' +
+      'Please add `aria-describedby` to the element this tooltip is being used to describe.');
+  }, []);
 
-  switch (direction) {
-    case 'top': classNames.push(styles.top); break;
-    case 'bottom': classNames.push(styles.bottom); break;
-    case 'left': classNames.push(styles.left); break;
-    case 'right': classNames.push(styles.right); break;
-    default: break;
-  }
+  // keep the div so enabling/disabling doesn't affect the dom tree or styles
+  if (disabled) return <div>{children}</div>;
```

Comment:
This is missing `styles.wrapper`. Not sure if that would affect the rendering of `children`.

Reply:
b46ecc0c